src/util/mask.js

```diff
@@ -90,14 +90,15 @@ const maskValidates = (mask, char) => {
  * @return {String}
  */
 export const maskText = (text, masked, dontFillMaskBlanks) => {
-  if (text == null) return ''
-  text = String(text)
```

Comment:
These lines were here for a reason. The value should still be cast to a string before calculating the cursor position, because that's what's going to be set on the input in the end.

Reply:
I think accepting null is fine, but completely removing this means that the cursor position is wrong if the mask changes when the value is falsy (`0` with `v-model.number` for example`). You should be casting `newText` to string (with a *proper* null check) in `updateRange()` and `watch.mask()` before the loop. This will also allow the removal of the ifs, because `for..of` on an empty string just gets skipped.

Reply:
I can see your point now. All codes that requires the return value of `maskText()` are expecting a string. So, converting the falsy to `'0'` is indeed correct. I'm reverting to your last commit.

Reply:
```js
if (text == null) return text
text = String(text)
```
Might work

Reply:
Well, now I prefer your last commit. Like I said, all readers of `maskText()` including `newValue` and `domProps.value` are expecting string.